sunpy/net/dataretriever/tests/test_norh.py

```diff
@@ -1,41 +1,144 @@
-])
-def test_get_url_for_time_range(timerange, url_start, url_end):
-    urls = norh.NoRHClient()._get_url_for_timerange(timerange, wavelength=17*u.GHz)
+def test_fetch_working():
+    """
+    Tests if the mock fetch contains the correct data.
+    """
+
+    qr1 = NORHClient.search(a.Time('2012/10/4', '2012/10/6'),
+                            a.Instrument('norh'), a.Wavelength(17*u.GHz))
+
+    # Create a mock query object
+    mock_qr = mock_query_object('2012/10/4', '2012/10/6', wavelength=17*u.GHz)
+
+    mock_qr = mock_qr[0]
+    qr = qr1[0]
+    # Assert the values
+    assert mock_qr.source == qr.source
+    assert mock_qr.instrument == qr.instrument
+    assert mock_qr.physobs == qr.physobs
+    assert mock_qr.provider == qr.provider
+    assert mock_qr.url == qr.url
+    assert mock_qr.time == qr.time
+
+    # Assert if the time range is same
+    assert qr1.time_range() == TimeRange('2012/10/4', '2012/10/7')
+
+    # Assert the fetch object, and whether it returns the correct set of files
+    res = NORHClient.fetch(qr1)
+    download_list = res
+    assert len(download_list) == len(qr1)
+
+    download_list.sort()
+    # Assert each of the files have the same file names
+    assert download_list[0].split('/')[-1].split('.')[0] == 'tca121004'
+    assert download_list[1].split('/')[-1].split('.')[0] == 'tca121005'
+    assert download_list[2].split('/')[-1].split('.')[0] == 'tca121006'
+
+
+@pytest.fixture
+def create_mock_url(mocker, sdate, edate, wave):
+    mocker.patch('sunpy.net.dataretriever.sources.norh.NoRHClient._get_url_for_timerange',
+                 return_value=create_url(sdate, edate, wave))
+
+
+@pytest.fixture
+def create_mock_search(mocker, sdate, edate, wave):
+    mocker.patch('sunpy.net.dataretriever.sources.norh.NoRHClient.search', side_effect=(
+           UnifiedResponse(mock_query_object(sdate, edate, wave))))
+
+
+@pytest.mark.usefixtures('create_mock_url')
+@pytest.mark.parametrize(
+    "sdate, edate, wave",
+    [('2012/3/7', '2012/3/14', 17*u.GHz),
+     ('2012/3/7', '2012/3/14', 34*u.GHz)]
+)
+def test_get_url_for_time_range(sdate, edate, wave):
+    urls = norh.NoRHClient()._get_url_for_timerange(TimeRange(parse_time('2012/3/7'),
+                                                              parse_time('2012/3/14')),
+                                                    wavelength=wave)
     assert isinstance(urls, list)
-    assert urls[0] == url_start
-    assert urls[-1] == url_end
+    if wave.value == 17.0:
+        assert urls[0] == 'ftp://anonymous:data@sunpy.org@solar-pub.nao.ac.jp/pub/nsro/norh/data/tcx/2012/03/tca120307'
+        assert urls[-1] == 'ftp://anonymous:data@sunpy.org@solar-pub.nao.ac.jp/pub/nsro/norh/data/tcx/2012/03/tca120314'
+    else:
+        assert urls[0] == 'ftp://anonymous:data@sunpy.org@solar-pub.nao.ac.jp/pub/nsro/norh/data/tcx/2012/03/tcz120307'
+        assert urls[-1] == 'ftp://anonymous:data@sunpy.org@solar-pub.nao.ac.jp/pub/nsro/norh/data/tcx/2012/03/tcz120314'
```

Comment:
To avoid repeated text, could we have:
```python
if wave == 17:
    prefix_filename = 'tca'
else:
    prefix_filename = 'tcz'
```
and keep the two asserts as:
```python
assert urls[0] == f'ftp://anonymous:data@sunpy.org@solar-pub.nao.ac.jp/pub/nsro/norh/data/tcx/2012/03/{prefix_filename}120307'
```

Reply:
Done.